Logstash/logstash-input-azuretopic/lib/logstash/inputs/azuretopic.rb

```diff
@@ -27,16 +28,21 @@ def initialize(*args)
   def register
     Azure.configure do |config|
       config.sb_namespace = @namespace
-      config.sb_access_key = @access_key
+      config.sb_sas_key_name = @access_key_name
+      config.sb_sas_key = @access_key
     end
-    @azure_service_bus = Azure::ServiceBus::ServiceBusService.new
+    signer = Azure::ServiceBus::Auth::SharedAccessSigner.new
+    sb_host = "https://#{Azure.sb_namespace}.servicebus.windows.net"
+
+    @azure_service_bus = Azure::ServiceBus::ServiceBusService.new(sb_host, { signer: signer})
   end # def register
 
   def process(output_queue)
     message = @azure_service_bus.receive_subscription_message(@topic ,@subscription, { :peek_lock => true, :timeout => 1 } )
     if message
       codec.decode(message.body) do |event|
         decorate(event)
+        event.set("type", @topic)
```

Comment:
Could you please explain why do we set this hard-coded property of type to the event? What if there's already property named 'type' in the log?